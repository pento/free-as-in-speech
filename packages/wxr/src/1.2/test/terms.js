/**
 * External dependencies
 */
const FDBFactory = require( 'fake-indexeddb/lib/FDBFactory' );

/**
 * Internal dependencies
 */
const { WXRDriver } = require( '../index.js' );

describe( 'Terms and Taxonomies', () => {
	beforeEach( () => {
		window.indexedDB = new FDBFactory();
	} );

	test( 'All Category fields are written as expected', async () => {
		const wxr = new WXRDriver();
		await wxr.connect();

		wxr.addCategory( {
			id: 1,
			slug: 'some-cat',
			name: 'Some Category',
			description: 'A category of things.',
			meta: [
				{ key: 'a-key', value: 'a-value' },
				{ key: 'another-key', value: 'another-value' },
			],
		} );

		wxr.addCategory( {
			id: 2,
			slug: 'some-other-cat',
			parent: 'some-cat',
			name: 'Some Other Category',
			description: 'A different category of things.',
		} );

		const xml = await wxr.export();

		expect( xml ).toMatchSnapshot();
	} );

	test( 'All Tag fields are written as expected', async () => {
		const wxr = new WXRDriver();
		await wxr.connect();

		wxr.addTag( {
			id: 1,
			slug: 'some-tag',
			name: 'Some Tag',
			description: 'A tag for things.',
			meta: [
				{ key: 'a-key', value: 'a-value' },
				{ key: 'another-key', value: 'another-value' },
			],
		} );

		wxr.addTag( {
			id: 2,
			slug: 'some-other-tag',
			name: 'Some Other Tag',
			description: 'A different tag for things.',
		} );

		const xml = await wxr.export();

		expect( xml ).toMatchSnapshot();
	} );

	test( 'All custom taxonomy fields are written as expected', async () => {
		const wxr = new WXRDriver();
		await wxr.connect();

		wxr.addTerm( {
			id: 1,
			taxonomy: 'my-tax',
			slug: 'some-term',
			name: 'Some Term',
			description: 'A Term in a Taxonomy of things.',
			meta: [
				{ key: 'a-key', value: 'a-value' },
				{ key: 'another-key', value: 'another-value' },
			],
		} );

		wxr.addTerm( {
			id: 2,
			taxonomy: 'my-tax',
			slug: 'some-other-term',
			name: 'Some Other Term',
			description: 'A different Term in the same Taxonomy of things.',
		} );

		wxr.addTerm( {
			id: 3,
			taxonomy: 'my-other-tax',
			slug: 'some-term',
			name: 'Some Term',
			description: 'The same Term in a different Taxonomy of things.',
		} );

		const xml = await wxr.export();

		expect( xml ).toMatchSnapshot();
	} );
} );
