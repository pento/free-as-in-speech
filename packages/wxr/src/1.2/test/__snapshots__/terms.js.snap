// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terms and Taxonomies All Category fields are written as expected 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>

<!-- This is a WordPress eXtended RSS file generated by the \\"Free (as in Speech)\\" browser extension, as an export of your site. -->
<!-- It contains information about your site's posts, pages, comments, categories, and other content. -->
<!-- You may use this file to transfer that content from one site to another. -->
<!-- This file is not intended to serve as a complete backup of your site. -->

<!-- To import this information into a WordPress site follow these steps: -->
<!-- 1. Log in to that site as an administrator. -->
<!-- 2. Go to Tools: Import in the WordPress admin panel. -->
<!-- 3. Install the \\"WordPress\\" importer from the list. -->
<!-- 4. Activate & Run Importer. -->
<!-- 5. Upload this file using the form provided on that page. -->
<!-- 6. You will first be asked to map the authors in this export file to users -->
<!--    on the site. For each author, you may choose to map to an -->
<!--    existing user on the site or to create a new user. -->
<!-- 7. WordPress will then import each of the posts, pages, comments, categories, etc. -->
<!--    contained in this file into your site. -->

<rss version=\\"2.0\\"
	xmlns:excerpt=\\"http://wordpress.org/export/1.2/excerpt/\\"
	xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\"
	xmlns:wfw=\\"http://wellformedweb.org/CommentAPI/\\"
	xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\"
	xmlns:wp=\\"http://wordpress.org/export/1.2/\\"
>
<channel>
	<wp:category>
		<wp:term_id>1</wp:term_id>
		<wp:category_nicename><![CDATA[some-cat]]></wp:category_nicename>
		<wp:category_parent><![CDATA[]]></wp:category_parent>
		<wp:cat_name><![CDATA[Some Category]]></wp:cat_name>
		<wp:category_description><![CDATA[A category of things.]]></wp:category_description>
		<wp:term_meta>
			<wp:meta_key>a-key</wp:meta_key>
			<wp:meta_value>a-value</wp:meta_value>
		</wp:term_meta>
		<wp:term_meta>
			<wp:meta_key>another-key</wp:meta_key>
			<wp:meta_value>another-value</wp:meta_value>
		</wp:term_meta>
	</wp:category>
	<wp:category>
		<wp:term_id>2</wp:term_id>
		<wp:category_nicename><![CDATA[some-other-cat]]></wp:category_nicename>
		<wp:category_parent><![CDATA[some-cat]]></wp:category_parent>
		<wp:cat_name><![CDATA[Some Other Category]]></wp:cat_name>
		<wp:category_description><![CDATA[A different category of things.]]></wp:category_description>
	</wp:category>
</channel>
</rss>"
`;

exports[`Terms and Taxonomies All Tag fields are written as expected 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>

<!-- This is a WordPress eXtended RSS file generated by the \\"Free (as in Speech)\\" browser extension, as an export of your site. -->
<!-- It contains information about your site's posts, pages, comments, categories, and other content. -->
<!-- You may use this file to transfer that content from one site to another. -->
<!-- This file is not intended to serve as a complete backup of your site. -->

<!-- To import this information into a WordPress site follow these steps: -->
<!-- 1. Log in to that site as an administrator. -->
<!-- 2. Go to Tools: Import in the WordPress admin panel. -->
<!-- 3. Install the \\"WordPress\\" importer from the list. -->
<!-- 4. Activate & Run Importer. -->
<!-- 5. Upload this file using the form provided on that page. -->
<!-- 6. You will first be asked to map the authors in this export file to users -->
<!--    on the site. For each author, you may choose to map to an -->
<!--    existing user on the site or to create a new user. -->
<!-- 7. WordPress will then import each of the posts, pages, comments, categories, etc. -->
<!--    contained in this file into your site. -->

<rss version=\\"2.0\\"
	xmlns:excerpt=\\"http://wordpress.org/export/1.2/excerpt/\\"
	xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\"
	xmlns:wfw=\\"http://wellformedweb.org/CommentAPI/\\"
	xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\"
	xmlns:wp=\\"http://wordpress.org/export/1.2/\\"
>
<channel>
	<wp:tag>
		<wp:term_id>1</wp:term_id>
		<wp:tag_slug><![CDATA[some-tag]]></wp:tag_slug>
		<wp:tag_name><![CDATA[Some Tag]]></wp:tag_name>
		<wp:tag_description><![CDATA[A tag for things.]]></wp:tag_description>
		<wp:term_meta>
			<wp:meta_key>a-key</wp:meta_key>
			<wp:meta_value>a-value</wp:meta_value>
		</wp:term_meta>
		<wp:term_meta>
			<wp:meta_key>another-key</wp:meta_key>
			<wp:meta_value>another-value</wp:meta_value>
		</wp:term_meta>
	</wp:tag>
	<wp:tag>
		<wp:term_id>2</wp:term_id>
		<wp:tag_slug><![CDATA[some-other-tag]]></wp:tag_slug>
		<wp:tag_name><![CDATA[Some Other Tag]]></wp:tag_name>
		<wp:tag_description><![CDATA[A different tag for things.]]></wp:tag_description>
	</wp:tag>
</channel>
</rss>"
`;

exports[`Terms and Taxonomies All custom taxonomy fields are written as expected 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>

<!-- This is a WordPress eXtended RSS file generated by the \\"Free (as in Speech)\\" browser extension, as an export of your site. -->
<!-- It contains information about your site's posts, pages, comments, categories, and other content. -->
<!-- You may use this file to transfer that content from one site to another. -->
<!-- This file is not intended to serve as a complete backup of your site. -->

<!-- To import this information into a WordPress site follow these steps: -->
<!-- 1. Log in to that site as an administrator. -->
<!-- 2. Go to Tools: Import in the WordPress admin panel. -->
<!-- 3. Install the \\"WordPress\\" importer from the list. -->
<!-- 4. Activate & Run Importer. -->
<!-- 5. Upload this file using the form provided on that page. -->
<!-- 6. You will first be asked to map the authors in this export file to users -->
<!--    on the site. For each author, you may choose to map to an -->
<!--    existing user on the site or to create a new user. -->
<!-- 7. WordPress will then import each of the posts, pages, comments, categories, etc. -->
<!--    contained in this file into your site. -->

<rss version=\\"2.0\\"
	xmlns:excerpt=\\"http://wordpress.org/export/1.2/excerpt/\\"
	xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\"
	xmlns:wfw=\\"http://wellformedweb.org/CommentAPI/\\"
	xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\"
	xmlns:wp=\\"http://wordpress.org/export/1.2/\\"
>
<channel>
	<wp:term>
		<wp:term_id>1</wp:term_id>
		<wp:tag_taxonomy><![CDATA[my-tax]]></wp:tag_taxonomy>
		<wp:term_slug><![CDATA[some-term]]></wp:term_slug>
		<wp:term_parent><![CDATA[]]></wp:term_parent>
		<wp:term_name><![CDATA[Some Term]]></wp:term_name>
		<wp:term_description><![CDATA[A Term in a Taxonomy of things.]]></wp:term_description>
		<wp:term_meta>
			<wp:meta_key>a-key</wp:meta_key>
			<wp:meta_value>a-value</wp:meta_value>
		</wp:term_meta>
		<wp:term_meta>
			<wp:meta_key>another-key</wp:meta_key>
			<wp:meta_value>another-value</wp:meta_value>
		</wp:term_meta>
	</wp:term>
	<wp:term>
		<wp:term_id>2</wp:term_id>
		<wp:tag_taxonomy><![CDATA[my-tax]]></wp:tag_taxonomy>
		<wp:term_slug><![CDATA[some-other-term]]></wp:term_slug>
		<wp:term_parent><![CDATA[]]></wp:term_parent>
		<wp:term_name><![CDATA[Some Other Term]]></wp:term_name>
		<wp:term_description><![CDATA[A different Term in the same Taxonomy of things.]]></wp:term_description>
	</wp:term>
	<wp:term>
		<wp:term_id>3</wp:term_id>
		<wp:tag_taxonomy><![CDATA[my-other-tax]]></wp:tag_taxonomy>
		<wp:term_slug><![CDATA[some-term]]></wp:term_slug>
		<wp:term_parent><![CDATA[]]></wp:term_parent>
		<wp:term_name><![CDATA[Some Term]]></wp:term_name>
		<wp:term_description><![CDATA[The same Term in a different Taxonomy of things.]]></wp:term_description>
	</wp:term>
</channel>
</rss>"
`;
